test/metabase/models/database_test.clj

```diff
@@ -0,0 +1,33 @@
+(ns metabase.models.database-test
+  (:require [clojure
+             [string :refer [join lower-case split]]
+             [test :refer :all]]
+            [metabase.models :refer [Database]]
+            [toucan.db :as db]))
+
+(defn- create-database!
+  [db-name engine]
+  (db/insert! Database
+        :name db-name
+        :engine (name engine)
+        :details {:db (join "-" (mapv lower-case (split db-name #"(?=[A-Z])")))}))
+
+(deftest disallow-duplicate-databases-with-same-name-and-engine
```

Comment:
we should only run these if we're currently testing against that database, for example we want to be able to run tests against just `:h2` without having a MySQL or Postgres database running. What we usually do is 

```clj
(require '[metabase.driver :as driver] '[metabase.test :as mt])

(deftest my-test
  ;; test-drivers will run the body against :mysql, :postgres, etc. if tests are enabled for that DB
  (mt/test-drivers #{:mysql :postgres :h2}
    ;; driver/*driver* will be bound to :mysql, :postgres, etc.
    (testing driver/*driver*
      ...)))
```

Reply:
actually on second thought you don't really to test this against the different DB types, you can just write the test once, we run tests with all of our supported backing application database types anyway. But you should clear out the DB when you're done. Maybe a test like

```clj
(deftest my-test
  (try
    (create-database! "My DB")
    (is (thrown? ...))
    (finally
      (delete-database "My DB"))))
```

that should probably do it